### ISSUE TITLE
Repo thread_local link error: undefined reference to TLS init function.

### ISSUE THREAD

**Issue by MaggieYingYi** [SNSystems/llvm-project-prepo#62]
Consider three files: foo.hpp, a.cpp and b.cpp, listed below.

$ cat foo.hpp
```
class foo
{
  public:
  inline  foo() {}
  inline ~foo() {}
  inline double bar() { return 88.88; }
};
```
$ cat a.cpp
```
#include "foo.hpp"
thread_local foo foo_instance;
```
$ cat b.cpp
```
#include "foo.hpp"
extern thread_local foo foo_instance;
int main(int argc, char** argv)
{
  double bar = foo_instance.bar();
  return 0;
}
```

Compile using:
```
$clang++ --target=x86_64-pc-linux-gnu-repo a.cpp -c -o a.repo.o -std=c++11
$clang++ --target=x86_64-pc-linux-gnu-repo b.cpp -c -o b.repo.o -std=c++11
$repo2obj a.repo.o -o a.elf.o
$repo2obj b.repo.o -o b.elf.o
$clang++ a.elf.o b.elf.o -o test
```

a.elf.o: In function `TLS wrapper function for foo_instance':
(.text+0x95): undefined reference to `TLS init function for foo_instance'
b.elf.o: In function `TLS wrapper function for foo_instance':
(.text+0x35): undefined reference to `TLS init function for foo_instance'
b.elf.o: In function `TLS wrapper function for foo_instance':
(.text+0x48): undefined reference to `TLS init function for foo_instance'
clang-8: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by MaggieYingYi** [comment]
Under code review: https://github.com/SNSystems/llvm-project-prepo/pull/63

**Comment by paulhuggett** [comment]
I'd be really in understanding some of the background to the fix proposed in https://github.com/SNSystems/llvm-project-prepo/pull/63. For example, the code supports an alias with a particular name: would this generalize? What about TLS for other targets, I'd guess they could use different names. Do aliases see other usage? (We've previously encountered then in IA64 ctor/dtor variants.)

**Comment by MaggieYingYi** [comment]
> 
> 
> I'd be really in understanding some of the background to the fix proposed in #63. For example, the code supports an alias with a particular name: would this generalize? What about TLS for other targets, I'd guess they could use different names. Do aliases see other usage? (We've previously encountered then in IA64 ctor/dtor variants.)

I have updated the #63 to enable the repo to support the aliases, which include the previous ctor/dtor variants. After code review #63 and merge it back to the master branch, we could revert previous short-term fix (https://github.com/SNSystems/llvm-project-prepo/issues/3#issuecomment-530820794).

**Comment by MaggieYingYi** [comment]
The issue has been fixed in the commit in 1077a252. 
 For the follow-up work, a new issue #64 has been created.
